common/model/cpptransaction.cc

```diff
@@ -0,0 +1,241 @@
+/*
+ * Copyright (c) 2018 IOTA Stiftung
+ * https://github.com/iotaledger/entangled
+ *
+ * Refer to the LICENSE file for licensing information
+ */
+
+#include "common/model/cpptransaction.h"
+#include "common/trinary/trit_array.h"
+
+namespace model {
+transaction::transaction() {
+  iota_transaction = transaction_new();
+}
+
+transaction::transaction(std::string trytes) {
+  iota_transaction = transaction_new();
+  size_t to_len = trytes.size() * 3;
+  size_t flex_len = flex_trits_num_for_trits(to_len);
+  flex_trit_t trits[flex_len];
+  tryte_to_flex_trit(trits, to_len, (const tryte_t *)trytes.data(), trytes.size(), trytes.size());
+  transaction_deserialize_from_trits(iota_transaction, trits);
+}
+
+transaction::~transaction() {
+  transaction_free(iota_transaction);
+}
+
+std::string transaction::get_signature(void) {
+  auto trits = transaction_signature(iota_transaction);
```

Comment:
assuming that we want to return a `string` here (which I'm not so sure of), you should much rather first create a fixed-length string and then write directly to it instead of copying twice